.gitignore

```diff
@@ -1,5 +1,8 @@
 .DS_Store
 
+.idea/
+.vscode/
```

Comment:
Could you explain why we need to remove these directory?

Reply:
I remember that these directories are sometimes auto-generated by the IDE, which can lead to unintended changes in the repository. While we need to block them, there are also specific settings that we include deliberately. I'll review what our `.gitignore` file needs to include shortly.

Reply:
Yeah if you dont ignore them, they will just keep showing up in random prs when people `git add .`

Reply:
exactly
Each developer can keep their IDE settings how they like, they don't need to be checked into the repository in my opinion